stripe-core/src/main/java/com/stripe/android/core/utils/IsWorkManagerAvailable.kt

```diff
@@ -0,0 +1,22 @@
+package com.stripe.android.core.utils
+
+import androidx.annotation.RestrictTo
+import androidx.work.WorkManager
+
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+fun interface IsWorkManagerAvailable {
+    operator fun invoke(): Boolean
+}
+
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+object RealIsWorkManagerAvailable : IsWorkManagerAvailable {
+    override fun invoke(): Boolean {
+        val workManagerInClasspath = runCatching {
+            Class.forName("androidx.work.WorkManager")
+        }.isSuccess
+
+        return FeatureFlags.workManagerForAnalytics.isEnabled &&
+            workManagerInClasspath &&
+            WorkManager.isInitialized()
```

Comment:
Not needed for this PR, but maybe worth logging wether or not work manager is available on devices when making this check? So we can better decide if we want to directly include the dependency in the future.

Reply:
yep I think that's enough!